src/screens/LoginScreen.js

```diff
@@ -1,50 +1,46 @@
 // 3rd part libraries - core
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {View, Text, Keyboard} from 'react-native';
+import {View, Text, AsyncStorage} from 'react-native';
 import {connect} from 'react-redux';
-import { Button, FormInput, FormLabel } from 'react-native-elements'
+import {Button, Card} from 'react-native-elements'
 // 3rd party libraries - additional
+import _ from 'lodash';
 
 // actions
-import {emailChanged, passwordChanged, loginUser} from '../actions/index';
+import {checkAndSetToken, setGuestToken, logUserIntoFacebook} from '../actions/index';
 
 // language and styles
 import {textDark, lightwhiteblue, facebookBlue, grayblue, navyBlue} from '../styles/index';
 import {login} from '../locale.en';
+import {navKeys} from '../constants';
+
+const {campo_libre, tagline, login_as_guest, login_with_facebook} = login;
 
 
 // our components - core
 // our components - additional
-import {  Input, Card, CardSection, PasswordInput} from '../components/common/index';
-import ListScreen from "../components/SearchList";
-import MapScreen from "../components/SearchMap";
-
 
 class LoginScreen extends Component {
-    state = {
-        hidden: true
-    };
+
+    componentWillMount() {
+        this.props.checkAndSetToken();
+    }
```

Comment:
Before the app mounts, check the token's status. If there is one, let's grab it and put it on the global state. If there isn't, let's note that, and either way when we're done, make sure to say that the app is ready so we hide the splash screen.